### PR TITLE
feat: reflect the changes immediately and refresh when close view modal

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
Please see order by commits!!!

- Reflect the changes **immediately** whenever the widget form is modified.
- refresh widget when close the view modal

https://github.com/cloudforet-io/console/assets/18563857/796c5550-2087-4d2c-bc92-f1fc1d4a82b8

### CODE REVIEW

File: apps/web/src/services/dashboards/widgets/_hooks/use-widget-lifecycle.ts [hunk 1]
```diff
@@ -174,11 +177,13 @@ export const useWidgetLifecycle = ({
     });
     watch(() => dashboardDetailState.variables, (after, before) => {
         if (!props.initiated || props.errorMode || !props.inheritOptions) return;
+        if (dashboardDetailState.widgetViewModeModalVisible && (props.widgetKey !== widgetFormState.widgetKey)) return; // disable when it's view mode
```
Comment (yuda110) [inline apps/web/src/services/dashboards/widgets/_hooks/use-widget-lifecycle.ts:180]: description: Without this, **all** widgets will be refreshed **every time** the dashboard variable is modified in the view modal.

File: apps/web/src/services/dashboards/shared/dashboard-widget-container/DashboardWidgetContainer.vue [hunk 2]
```diff
@@ -155,6 +160,10 @@ export default defineComponent<Props>({
                 refreshAllWidget();
             }
         });
+        const handleRefreshWidget = (widgetKey: string) => {
+            const targetWidgetRef = state.widgetRef.find((d) => d?.$el?.id === widgetKey);
+            targetWidgetRef?.refreshWidget();
```
Comment (yuda110) [inline apps/web/src/services/dashboards/shared/dashboard-widget-container/DashboardWidgetContainer.vue:165]: description: Refresh only the widget displayed in view modal.

**Review by WANZARGEN — APPROVED** [review]
LGTM!
It was very easy to read since you split commits in reasonable unit. :) Thank u~

**Review by skdud4659 — APPROVED** [review]
3728379B-3C22-491E-A813-822A273E693A
LGTM!

**Pull request merged**